File: 진우/11weeks/7785. 회사에 있는 사람/회사에 있는 사람.cc
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <set>

using namespace std;

int main() {
    int n;
    cin >> n;

    set<string> work;  
    // 회사에 "현재" 남아 있는 사람들의 이름을 저장하는 집합
    // set → 자동 정렬 + 중복 없음

    string name, log;
    for (int i = 0; i < n; i++) {
        cin >> name >> log;  
        // name : 이름
        // log  : enter 또는 leave

        if (log == "enter") {
            work.insert(name);  
            // 회사에 들어왔으므로 set에 추가
        }
        else {
            work.erase(name);   
            // leave → 회사에서 나갔으므로 set에서 삭제
        }
    }

    // 역순으로 출력하기 위해 vector로 옮김
    vector<string> people(work.begin(), work.end());

    // 내림차순 문자열 정렬
    sort(people.begin(), people.end(), greater<string>());

    // 결과 출력
    for (int i = 0; i < (int)people.size(); i++) {
        cout << people[i] << "\n";
    }

    return 0;
}

File: 진우/11weeks/7785. 회사에 있는 사람/회사에 있는 사람.py
n = int(input())  # 출입 기록의 개수 입력

work = set()  # 현재 회사 안에 있는 사람들을 저장하는 집합 (중복 없음)

for _ in range(n):
    name, log = input().split()  # 이름과 출입 기록(enter/leave) 입력받기

    if log == "enter":
        work.add(name)      # enter → 회사에 들어옴 → 집합에 추가
    else:
        work.remove(name)   # leave → 회사에서 나감 → 집합에서 제거

# set은 순서가 없기 때문에 정렬을 위해 리스트로 변환
people = list(work)

# 문자열 내림차순 정렬 (사전 역순)
people.sort(reverse=True)

# 출력
for i in range(len(people)):
    print(people[i])
